lib/GenomeFileUtil/core/FastaGFFToGenome.py

```diff
@@ -805,6 +855,47 @@ def _gen_genome_info(self, core_genome_name, scientific_name, assembly_ref,
                 if 'exon' in feature:
                     self._update_from_exons(feature)
                 genome['non_coding_features'].append(feature)
+
+            # Test if location order is in order.
+            is_transpliced = False
+            if "flags" in feature and "trans_splicing" in feature["flags"]:
+                is_transpliced = True      
+            if not is_transpliced and len(feature["location"]) > 1:
+                # Check the order only if not trans_spliced and has more than 1 location.    
+                location_warning = self._check_location_order(feature["location"])
+                if location_warning is not None:
+                    feature["warnings"] = feature.get('warnings', []) + [location_warning]     
+
+            # Test for full contig length features and if on both strands.
+            feature_min_location = None
+            feature_max_location = None
+            location_contigs = set()
+            strand_set = set()
+            #Only test if all locations on the same contig.
+            for location in feature["location"]:
+                location_contigs.add(location[0])
+            if len(location_contigs) == 1:
+                for location in feature["location"]:
+                    if location[2] == "+":
+                        strand_set.add("+")
+                        location_min = location[1]  
+                        location_max = location[1] + location[3] - 1
+                    if location[2] == "-":
+                        strand_set.add("+")
```

Comment:
I think you mean "-"

Reply:
Yeah you are right. Confused now on why it threw the warning. If I add "+", then "+" to a set, it only contains 1 "+" correct?